lib/docker/template/cache.rb

```diff
@@ -42,9 +42,10 @@ def aliased_context(builder)
       # --
 
       def cleanup(repo)
+        return unless repo.clean_cache?
         cache_dir = repo.cache_dir.parent
 
-        if repo.cacheable? && cache_dir.exist?
+        if cache_dir.exist?
```

Comment:
The repo.cacheable? should probably still be there.  The problem with removing that is we might end up trying to cleanup a rootfs image, which will end up causing an error, does adding that back in cause any problems?